src/components/network-chart/NetworkChartRow.js

```diff
@@ -34,7 +34,12 @@ export type NetworkChartRowBarProps = {
   +marker: TracingMarker,
   // Pass the payload in as well, since our types can't express a TracingMarker with
   // a specific payload.
-  +networkPayload: NetworkPayload,
+  markerMeta: {
```

Comment:
This type definition is repeated a few times. I think it would be better to collect it somewhere, like `src/types/profile-derived.js`. Also, the type definition should use an exact object and read-only properties, like so:

```diff
diff --git a/src/types/profile-derived.js b/src/types/profile-derived.js
index dbf0bde7..82d8eb15 100644
--- a/src/types/profile-derived.js
+++ b/src/types/profile-derived.js
@@ -126,3 +126,14 @@ export type LocalTrack =
 
 export type Track = GlobalTrack | LocalTrack;
 export type TrackIndex = number;
+
+/**
+ * The marker payloads need additional derived information for display. This information
+ * is collected on this type so that it can be passed around to various components.
+ */
+export type NetworkRequestInfo = {|
+  +requestQueue: number,
+  +request: number,
+  +response: number,
+  +type: string,
+|};
```